server/app/mutations/grid_certificates/common.rb

```diff
@@ -64,5 +69,84 @@ def validate_dns_record(domain, expected_record)
     rescue
       false
     end
+
+    # @param domain_authorization [GridDomainAuthorization]
+    def validate_domain_authorization(authz)
+
+      case authz.authorization_type
+      when 'dns-01'
+        # Check that the expected DNS record is already in place
+        unless check_dns_record(authz.domain, authz.challenge_opts['record_content'])
+          add_error(:dns_record, :invalid, "Expected DNS record not present for domain #{authz.domain}") # XXX: validations error type
+        end
+      end
+    end
+
+    # @param domains [Array<String>]
+    def validate_authorizations_for_domains(domains)
+      domains.each do |domain|
+        unless domain_authorization = get_authz_for_domain(self.grid, domain)
+          add_error(:authorization, :not_found, "Domain authorization not found for domain #{domain}") # XXX: validations error type
```

Comment:
The mutations error should be related to the input fields... so these should really be like `add_error(:domains, :domain_authorization, "...")`